packages/lib/reducer.ts

```diff
@@ -883,6 +883,12 @@ const reducer = produce((draft: Draft<State> = defaultState, action: any) => {
 			break;
 		}
 
+		case 'INITIAL_SELECTION_SET':
+			// To allow creating notes when opening the app with all notes and/or tags,
+			// we also need a "last selected folder ID".
+			draft.selectedFolderId ??= draft.settings.activeFolderId;
+			break;
```

Comment:
It could be simpler to use existing events:
```suggestion
		case 'SMART_FILTER_SELECT':
		case 'TAG_SELECT':
			// To allow creating notes when opening the app with all notes and/or tags,
			// we also need a "last selected folder ID".
			draft.selectedFolderId ??= draft.settings.activeFolderId;
			break;
```
This would, however, require moving the logic to a separate `switch` statement.

Reply:
In general, the reducer actions are essentially setters - you provides some value and this is used to update the state. While in this case it's more like an event that has some side effect and I think we should avoid this pattern if possible (although it's present for a few actions).

If the goal here is to set the selected folder, isn't it possible to call `FOLDER_SELECT`?

Reply:
`FOLDER_SELECT` also sets `notesParentType` to `'Folder'` and clears `selectedNoteIds`. However, this could be fine if `FOLDER_SELECT` is called before the initial selection is set.

An alternative could be to handle this in `SETTING_UPDATE_ONE` or `SETTING_UPDATE_ALL`, which would remove the need for `dispatch({ type: 'SOME_TYPE_TO_SET_THE_INITIAL_SELECTION_HERE' })` in the initialization logic for mobile and desktop.